tools/code_format/rst_check.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/python3
+
+import multiprocessing
+import re
+import subprocess
+import sys
+
+import rstcheck
+
+
+# things we dont want to see in generated docs
+# TODO(phlax): move to .rstcheck.cfg when available
+RSTCHECK_GREP_FAIL = (" ref:", "\[\#comment")
+RSTCHECK_CONFIG=".rstcheck.cfg"
+RSTCHECK_COMMAND = ("rstcheck", "-r")
+
+
+def run_rstcheck():
+    """Return 0 on success.
+
+    this is copied from rstcheck to capture stderr
+
+    """
+    args = rstcheck.parse_args()
+
+    if not args.files:
+        return 0
+
+    with rstcheck.enable_sphinx_if_possible():
```

Comment:
im wondering if its possible to use importlib to assert that sphinx is available...

Reply:
and just enable, ill check...